src/features/archaeologist/discovery.ts

```diff
@@ -74,4 +80,31 @@ export async function initialisePeerDiscovery(browserNode: Libp2p, setArchs: (ar
   browserNode.pubsub.subscribe(archEnvConfigTopic);
 
   return browserNode;
-}
\ No newline at end of file
+}
+
+
+export async function confirmArweaveTransaction(data: { arch: Archaeologist, arweaveTxId: string, unencryptedShardHash: string }) {
+  const { arch, arweaveTxId, unencryptedShardHash } = data;
+  try {
+    const archConnection = discoveredArchs[arch.address].connection;
+
+    if (!archConnection) throw new Error('No connection to archaeologist');
```

Comment:
We will need to think through the workflow where this error is thrown. 

We will probably need a way to monitor these connections during this communication workflow (maybe on an interval in it's own routine) and communicate specifically which archaeologist has dropped so the webapp can either retry or inform the user that one of the archaeologists is unavailable. Then we will need to scope out what the "replace archaeologist" workflow looks like.

As an aside -- do you know what would make this `archConnection` null/undefined? For example, if the browser connects to an archaeologist, but then the archaeologist goes offline, will this connection object still be defined?
I'm hopeful it would, but if it doesn't, we may need some extra logic on the "liveness" interval check to see if the comms connection is healthy.

Reply:
Yeah this is something I was worried about too. For some reason, the node's on disconnect handler works completely fine when the browser disconnects, but it doesn't work the other way round.

Using this connection object to implement some sorta heartbeat mechanism is something I hadn't thought of at all -- that should work. Also, when the arch goes offline the object still remains defined, so I think that's good.